modules/options/src/main/scala/scala/build/options/ComputeVersion.scala

```diff
@@ -5,26 +5,29 @@ import com.github.plokhotnyuk.jsoniter_scala.macros.*
 import org.eclipse.jgit.api.Git
 import org.eclipse.jgit.lib.{Constants, Ref}
 
-import scala.build.Positioned
+import scala.build.{Position, Positioned}
 import scala.build.errors.{BuildException, MalformedInputError}
 import scala.io.Codec
 import scala.jdk.CollectionConverters.*
 import scala.util.{Success, Try, Using}
 
 sealed abstract class ComputeVersion extends Product with Serializable {
   def get(workspace: os.Path): Either[BuildException, String]
+
+  val positions: Seq[Position]
 }
 
 object ComputeVersion {
 
-  final case class Command(command: Seq[String]) extends ComputeVersion {
+  final case class Command(positions: Seq[Position], command: Seq[String]) extends ComputeVersion {
```

Comment:
```suggestion
final case class Command(command: Seq[String], positions: Seq[Position] = Nil) extends ComputeVersion {
```

Maybe it is worth adding positions as the last argument, so in some cases, we don't have to pass `Nil` explicitly. This suggestion refers to every class that extends from `ComputeVersion`.

Reply:
I placed the `positions` at the end of the argument list, since it is more logical this way, but I'd opt for keeping the positioning explicit with no default values. `positions = Nil` is not a typical case, rather everywhere there should be either Poisition.File or Position.CommandLine.